tests/jballerina-unit-test/src/test/resources/test-src/types/never/never-type-runtime.bal

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+function testNeverRuntime() {
+    error? e1 = trap test1();
+    assertEquality(true, e1 is error);
+    assertEquality("Error value is:true", (<error>e1).message());
+
+    error? e2 = trap test2();
+    assertEquality(true, e2 is error);
+    assertEquality("Error value is:true", (<error>e2).message());
+
+    error? e3 = trap test3();
+    assertEquality(true, e3 is error);
+    assertEquality("Error value is:true", (<error>e3).message());
+
+    error? e4 = trap test4();
+    assertEquality(true, e4 is error);
+    assertEquality("Error value is:true", (<error>e4).message());
+
+    error? e5 = trap test5();
+    assertEquality(true, e5 is error);
+    assertEquality("Error value is:true", (<error>e5).message());
+
+    error? e6 = trap test6();
+    assertEquality(true, e6 is error);
+    assertEquality("Error value is:true", (<error>e6).message());
+}
+
+function func1() returns never {
+    panic error("error!");
+}
+
+function func2() returns record {| never x; |} {
+    panic error("error!");
+}
+
+function test1() {
+    boolean b = <any> func1 is function () returns record {| never x; |};
+    panic error("Error value is:" + string `${b}`);
```

Comment:
Can't we just assert `assertEquality(true, b)` instead of panicking here and introducing a separate test function?

Let's also add some negative tests - target has only never, source has something that's not never.